alacritty_terminal/src/term/mod.rs

```diff
@@ -447,10 +447,14 @@ impl<'a> Iterator for RenderableCellsIter<'a> {
 
                     renderable_cell.inner = RenderableCellContent::Cursor(cursor_key);
 
-                    if let Some(color) = self.config.colors.cursor.cursor {
-                        renderable_cell.fg = color;
+                    if self.config.colors.cursor.cursor.is_some() {
+                        // `ansi::NamedColor::Cursor` is in sync with `cursor.cursor` color from a
+                        // config, but we can change cursor color with escape sequence during
+                        // runtime.
+                        renderable_cell.fg = *self.colors.index(ansi::NamedColor::Cursor);
                     }
```

Comment:
Did the previous implementation behave the same way and only apply the cursor colors when it is fixed in the config?

It would be possible to always set it to a fixed color once it's set to a specific color once, even if it was just inversed before, but I'd mimic the previous behavior here. Not sure what other terminals do either, but I'd think usually they'd probably try to respect the change cursor color escape.

Reply:
> Did the previous implementation behave the same way and only apply the cursor colors when it is fixed in the config?

Yes.

Reply:
Sounds good to me then.